diffusion/generate.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2022 MosaicML Diffusion authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Generate images from a model."""
+
+import operator
+from typing import List
+
+import hydra
+from composer import Algorithm, ComposerModel
+from composer.algorithms.low_precision_groupnorm import apply_low_precision_groupnorm
+from composer.algorithms.low_precision_layernorm import apply_low_precision_layernorm
+from composer.core import Precision
+from composer.utils import reproducibility
+from datasets import load_dataset
+from omegaconf import DictConfig
+from torch.utils.data import Dataset
+
+from diffusion.evaluation.generate_images import ImageGenerator
+
+
+def generate(config: DictConfig) -> None:
+    """Evaluate a model.
+
+    Args:
+        config (DictConfig): Configuration composed by Hydra
+    """
+    reproducibility.seed_all(config.seed)
+
+    # The model to evaluate
+    if not config.hf_model:
+        model: ComposerModel = hydra.utils.instantiate(config.model)
+    else:
+        model = config.model.name
+
+    tokenizer = model.tokenizer if hasattr(model, 'tokenizer') else None
+
+    # The dataset to use for evaluation
+
+    if config.hf_model:
```

Comment:
Is it possible to use hf datasets with non-hf models and vice versa? If so, using a `hf_model` flag to determine the dataset type might be confusing/limiting.

Reply:
added a new flag `hf_dataset` to delineate the difference